modules/selection/src/main/kotlin/com.r3.corda.lib.tokens.selection/memory/internal/ExternalIdIndexUtils.kt

```diff
@@ -0,0 +1,67 @@
+package com.r3.corda.lib.tokens.selection.memory.internal
+
+import net.corda.core.crypto.toStringShort
+import net.corda.core.node.ServiceHub
+import java.security.PublicKey
+import java.util.*
+
+/** Table names. */
+val publicKeyHashToExternalId = "pk_hash_to_ext_id_map"
+
+/** Column names. */
+val publicKeyHashToExternalId_externalId = "external_id"
+val publicKeyHashToExternalId_publicKeyHash = "public_key_hash"
+
+sealed class Holder {
```

Comment:
isn't this more like "selection method" or something like that?

Reply:
No, this is Holder of the token, it can be key or external id, I introduced TokenOnly later, that's why name is confusing probably.